federator/src/lib/Federator.js

```diff
@@ -158,6 +167,35 @@ module.exports = class Federator {
                     _typeId: typeId
                 } = log.returnValues;
 
+                const {
+                    mediumAmount,
+                    largeAmount              
+                } = await allowTokens.getLimits(tokenAddress);
+                
+                const BN = this.mainWeb3.utils.BN;
+                const mediumAmountBN = new BN(mediumAmount);
+                const largeAmountBN = new BN(largeAmount);
+                const amountBN = new BN(amount);
+
+                if(mediumAndSmall) {   
+                    // At this point we're processing blocks newer than largeAmountConfirmations
+                    // and older than smallAmountConfirmations
+                    if(amountBN.gte(largeAmountBN)) {
+                        const c = currentBlock - blockNumber;
+                        const rC = largeAmountConfirmations;
+                        this.logger.debug(`[large amount] Tx: ${transactionHash} ${amount} ${symbol} won't be proccessed yet ${c} < ${rC}`);
+                        continue;
+                    } else if(
+                        (amountBN.gte(mediumAmountBN)) &&
+                        (currentBlock - blockNumber <= mediumAmountConfirmations)
```

Comment:
If the currentBlock - blockNumber is equal to the mediumAmountConfirmations we consider it as enough blocks
```suggestion
                        (currentBlock - blockNumber < mediumAmountConfirmations)
```